### PR TITLE
Furkan/registration/api

### PR DESCRIPTION
**Pull request by FurkannOzbek**
The api includes routes for 

Register
Login
and user information

Added extra middleware to verify the token for restrict outside reach to the db

### CODE REVIEW

File: api/src/routers/application.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import express from "express";
+
+// This router can be deleted once you add your own router
+const application = express.Router();
+
+application.get("/", (req, res) => {
```
Comment (magdazelena) [inline api/src/routers/application.js:6]: what does this endpoint do? or is this just a placeholdeR?

File: api/src/routers/login.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import express from "express";
+import knex from "../database_client.js";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
+
+// This router can be deleted once you add your own router
```
Comment (magdazelena) [inline api/src/routers/login.js:6]: what does this comment mean? it's in many places :)

File: api/src/routers/application.js (hunk 1 above)
Comment (FurkannOzbek) [reply api/src/routers/application.js:6]: Oh thats not in a use right now, its for the future , sorry for confusion. I will remove it now and i'll add it back when its right time :)

File: api/src/routers/login.js (hunk 2 above)
Comment (FurkannOzbek) [reply api/src/routers/login.js:6]: That comes from template, I did focus too much coding so I forgot to delete some unnecesarry comments from the template,I will clean up right away :D

**Review by magdazelena — APPROVED** [review]
👏

**Comment by FurkannOzbek** [comment]
- Includes new endpoint `/api/registration`
- Includes new endpoint `/api/login`
- Includes new endpoint `/api/user:id`

**Pull request merged**